network/gossip/libp2p/middleware.go

```diff
@@ -306,12 +306,6 @@ func (m *Middleware) SendDirect(msg *message.Message, targetID flow.Identifier)
 		return fmt.Errorf("failed to send message to %s: %w", targetID.String(), err)
 	}
 
-	// flush the stream
-	err = bufw.Flush()
-	if err != nil {
-		return fmt.Errorf("failed to flush stream for %s: %w", targetID.String(), err)
-	}
-
 	// track the number of bytes that will be written to the wire for metrics
 	byteCount := bufw.Buffered()
```

Comment:
Any reason we don't just use `msg.Size()`? Are we certain that the stream doesn't auto flush?

Reply:
msg.Size() will actually work just fine. I will change the code to use msg.Size().

Reply:
I picked up the idea of flush from an libp2p example.

Reply:
ok - so the `flush` is indeed needed. If I remove it, nothing gets written to the stream failing the unit tests.